### PR TITLE
[WIP] Add support for writing logs to WhyLabs Observatory

### PR DESCRIPTION
**Pull request by phantomgoose**
Provide an easy way for a user to upload dataset profiles to the WhyLabs platform.

### CODE REVIEW

File: src/whylogs/app/config.py [hunk 1]
```diff
@@ -179,7 +179,7 @@ class WriterConfigSchema(Schema):
     Marshmallow schema for :class:`WriterConfig` class.
     """
 
-    type = fields.Str(validate=validate.OneOf(["local", "s3"]), required=True)
+    type = fields.Str(validate=validate.OneOf(["local", "s3", "songbird"]), required=True)
```
Comment (andyndang) [inline src/whylogs/app/config.py]:
```suggestion
    type = fields.Str(validate=validate.OneOf(["local", "s3", "whylabs"]), required=True)
```

**Review by andyndang — CHANGES_REQUESTED** [review]
Rename all `songbird` to `whylabs` or `whylabs_api`

File: src/whylogs/app/session.py [hunk 2]
```diff
@@ -55,9 +56,13 @@ def __init__(
         )
         self.with_rotation_time = with_rotation_time
         self.cache_size = cache_size
+        self.useSongbird = useSongbird
 
     def __enter__(self):
         # TODO: configure other aspects
+        if self.useSongbird:
+            from whylogs.songbird.wrapper import start_session
```
Comment (andyndang) [inline src/whylogs/app/session.py]: please rename all `songbird` references to `whylabs`

File: src/whylogs/app/writers.py [hunk 3]
```diff
@@ -386,6 +386,37 @@ def _write_protobuf(self, profile: DatasetProfile):
             f.write(protobuf.SerializeToString())
 
 
+class SongbirdWriter(Writer):
```
Comment (andyndang) [inline src/whylogs/app/writers.py]: Same

File: src/whylogs/songbird/wrapper.py [hunk 4]
```diff
@@ -0,0 +1,45 @@
+from google.protobuf.message import Message
+import whylabs_client
+from whylabs_client.api import sessions_api
+import requests
+
+configuration = whylabs_client.Configuration( host = "https://songbird.development.whylabsdev.com" )
```
Comment (andyndang) [inline src/whylogs/songbird/wrapper.py]: Should be `https://api.whylabsapp.com`

File: src/whylogs/songbird/wrapper.py [hunk 5]
```diff
@@ -0,0 +1,45 @@
...
+        return sessions_api.SessionsApi(api_client)
+
+def start_session():
+    global session_token
+    client = get_songbird_client()
+    try:
+        response = client.create_session()
+        session_token = response.get("token")
+    except Exception as e:
+        print(f"Failed to create songbird session: {e}")
```
Comment (andyndang) [inline src/whylogs/songbird/wrapper.py]: Don't use `printf`. Use `logger` API instead

File: src/whylogs/app/writers.py [hunk 6]
```diff
@@ -386,6 +386,37 @@ def _write_protobuf(self, profile: DatasetProfile):
...
+    def write(self, profile: DatasetProfile, rotation_suffix: str = None):
+        """
+        Write a dataset profile to Songbird
+        """
+        self.rotation_suffix = rotation_suffix
+
+        for fmt in self.formats:
```
Comment (andyndang) [inline src/whylogs/app/writers.py]: This writer should disregard other formats and only write protobuf

**Review by andyndang — CHANGES_REQUESTED** [review]

File: src/whylogs/app/session.py [hunk 7]
```diff
@@ -38,7 +38,8 @@ def __init__(
         writers: List[Writer],
         verbose: bool = False,
         with_rotation_time: str = None,
-        cache_size: int = None
+        cache_size: int = None,
+        useSongbird: bool = False,
```
Comment (lalmei) [inline src/whylogs/app/session.py]: shouldnt you check the writer list for whylabs writer ?

File: src/whylogs/app/session.py [hunk 8]
```diff
@@ -55,9 +56,13 @@ def __init__(
         )
         self.with_rotation_time = with_rotation_time
         self.cache_size = cache_size
+        self.useSongbird = useSongbird
```
Comment (lalmei) [inline src/whylogs/app/session.py]: as above

File: src/whylogs/app/session.py (hunk 7 above)
Comment (lalmei) [reply src/whylogs/app/session.py]: or this parameter add a whylabs writer

Comment (andyndang) [reply src/whylogs/app/session.py]: Yeah it's duplicated with `writers` up there. I don't think the parameter should be there

File: src/whylogs/songbird/wrapper.py (hunk 5 above)
Comment (phantomgoose) [reply src/whylogs/songbird/wrapper.py]: Replaced debug logging with `logger`, but what about printing the URL at the end of a session?

**Comment by lalmei** [comment]
@phantomgoose was this deprecated with #195

**Pull request closed without merging**